app/controllers/actions/RequiredUserAnswers.scala

```diff
@@ -49,66 +49,70 @@ class RequiredUserAnswers @Inject()(genericLogger: GenericLogger)(implicit val e
     }
   }
 
-  private[controllers] def returnMissingAnswers[A: ClassTag, B: ClassTag](list: List[RequiredPage[_,_,_]])
-                                                                         (implicit request: DataRequest[_]): List[RequiredPage[_,_,_]] = {
+  private[controllers] def returnMissingAnswers[A: ClassTag, B: ClassTag](list: List[RequiredPage[_, _, _]])
+                                                                         (implicit request: DataRequest[_]): List[RequiredPage[_, _, _]] = {
     list.filterNot { listItem =>
       val currentPage: Option[A] = request.userAnswers.get(listItem.pageRequired.asInstanceOf[QuestionPage[A]])(listItem.reads.asInstanceOf[Reads[A]])
       (currentPage.isDefined, listItem.basedOnPreviousPages.nonEmpty) match {
         case (false, true) =>
-           val userAnswersMatched:List[Boolean] = listItem.basedOnPreviousPages.map { previousListItem =>
+          val userAnswersMatched: List[Boolean] = listItem.basedOnPreviousPages.map { previousListItem =>
             val previousPage: PreviousPage[QuestionPage[B], B] = previousListItem.asInstanceOf[PreviousPage[QuestionPage[B], B]]
             val previousPageAnswer: Option[B] = request.userAnswers.get(previousPage.page)(previousPage.reads)
-            !previousPageAnswer.exists(i => previousPage.previousPageAnswerRequired.contains(i))
+            previousPage.previousPageAnswerRequired match {
+              case Nil => !previousPageAnswer.isDefined
+              case _ => !previousPageAnswer.exists(i => previousPage.previousPageAnswerRequired.contains(i))
+            }
           }
-          if(userAnswersMatched.contains(true) && userAnswersMatched.contains(false)) {
+          if (userAnswersMatched.contains(true) && userAnswersMatched.contains(false)) {
             true
           } else {
             !userAnswersMatched.contains(false)
           }
```

Comment:
I'm not sure what the purpose of this if statement is the first part is returning true if the userAnswersMatched contains both true and false and the else element is returning true if the userAnswersMatched contains false. So whether the userAnswersMatched contains true doesn't seem to matter and if ```
(userAnswersMatched.contains(true) && userAnswersMatched.contains(false)) {
            true
          } else {
            !userAnswersMatched.contains(false)
          }
``` could be replaced with `!userAnswersMatched.contains(false)`

Reply:
its if  1 previous answers is answered but the other one isnt, then we do not require the current page to be answered
userAnswersMatched.contains(true) <- previous user answer doesnt exist
userAnswersMatched.contains(false) <- previous user answer does exist

else
if it doesnt contain true, and only contains false (by virtue of the if statement) then we require the current user answer

Reply:
as its a list

Reply:
![image](https://github.com/hmrc/soft-drinks-industry-levy-registration-frontend/assets/26716179/299e1c2f-2f88-42f1-9a98-a47beba541cd)
brain hurter

Reply:
My point was it is returning true whether you check for if it contains true or not.

Reply:
We can leave it for now though given we are going to be having another story around this.